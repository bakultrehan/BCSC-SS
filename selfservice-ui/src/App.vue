<template>
  <v-app>
    <Header :hideMenu="hideMenu" />
    <a class="anchor" id="top"></a>
    <v-content>
      <v-container fluid class="main-content" id="main">
        <router-view />
      </v-container>
    </v-content>
    <Footer />
  </v-app>
</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import Header from '@/components/Header/Header.vue';
import Footer from '@/components/Footer/Footer.vue';

@Component({
  components: {
    Header,
    Footer
  }
})
export default class App extends Vue {
  private hideMenu: boolean = false;
  constructor() {
    super();
  }
  @Watch('$route', { immediate: true, deep: true })
  private onUrlChange(newVal: any) {
    const { hideMenu = false } = newVal.meta;
    this.hideMenu = hideMenu || false;
  }

  private created() {
    const { hideMenu = false } = this.$route.meta;
    this.hideMenu = hideMenu || false;
  }
}
</script>

<style lang="scss">
@import './assets/styles/theme.scss';

#app {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
}
.main-content {
  margin-top: 10px;
  @include sm {
    margin-top: 56px;
  }
}
a.anchor {
  display: block;
  position: relative;
  top: -250px;
  visibility: hidden;
}
</style>
