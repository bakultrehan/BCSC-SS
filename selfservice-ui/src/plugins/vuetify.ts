import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
