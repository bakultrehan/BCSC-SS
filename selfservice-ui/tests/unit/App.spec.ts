import { shallowMount, createLocalVue } from '@vue/test-utils';
import App from '@/App.vue';
import vuetify from 'vuetify';

describe('App.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(App, {
      localVue,
      stubs: ['router-view'],
      mocks: { $route: { meta: { showVerticalMenu: false } } }
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
