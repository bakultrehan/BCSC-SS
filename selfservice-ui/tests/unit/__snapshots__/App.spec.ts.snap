// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders props when passed 1`] = `
<v-app-stub
  id="app"
>
  <header-stub />
   
  <v-content-stub
    tag="main"
  >
    <v-container-stub
      class="main-content"
      fluid="true"
      id="main"
      tag="div"
    >
      <router-view-stub />
    </v-container-stub>
  </v-content-stub>
   
  <footer-stub />
</v-app-stub>
`;
